code/game/gamemodes/trifecta/trifecta.dm

```diff
@@ -101,12 +103,80 @@
 /datum/game_mode/trifecta/post_setup()
 	for(var/datum/mind/vampire as anything in pre_vampires)
 		vampire.add_antag_datum(/datum/antagonist/vampire)
+
 	for(var/datum/mind/changeling as anything in pre_changelings)
 		changeling.add_antag_datum(/datum/antagonist/changeling)
+
 	for(var/datum/mind/traitor as anything in pre_traitors)
-		traitor.add_antag_datum(/datum/antagonist/traitor)
+		var/datum/antagonist/traitor/tot_datum = new()
+		tot_datum.delayed_objectives = TRUE
+		traitor.add_antag_datum(tot_datum)
+
+	if(length(pre_traitors))
+		var/random_time = rand(300 SECONDS, 900 SECONDS)
+		addtimer(CALLBACK(src, PROC_REF(late_handout)), random_time)
+
 	..()
 
+/datum/game_mode/trifecta/proc/traitors_to_add()
+	var/extra_points = cost_at_roundstart - num_players()
+	if(extra_points - TOT_COST < 0)
+		return 0 // Not enough new players to add extra tots
+
+	. = 0
+	while(extra_points)
+		if(extra_points < TOT_COST)
+			.++
+			return
+		extra_points -= TOT_COST
+		.++
+
+/datum/game_mode/trifecta/late_handout()
+	var/traitors_to_add = 0
+
+	for(var/datum/mind/traitor_mind as anything in traitors)
+		if(QDELETED(traitor_mind) || !traitor_mind.current) // Explicitly no client check in case you happen to fall SSD when this gets ran
+			traitors_to_add++
+			traitors -= traitor_mind
+			continue
+		for(var/datum/antagonist/traitor/traitor_datum in traitor_mind.antag_datums)
+			for(var/datum/objective/objective as anything in traitor_datum.objective_holder.objectives)
+				objective.delayed_objective = FALSE
+				objective.target = null
+
+				var/datum/objective/steal/possible_steal_objective = objective
+				if(istype(possible_steal_objective))
+					possible_steal_objective.steal_target = null
+
+				traitor_datum.objective_holder.assigned_targets = list()
```

Comment:
same stuff here. All of this is duplicated code, you should probably put this on `/gamemode` and only have it called in certain gamemode's `late_handout`s